crates/bevy_core_pipeline/src/oit/mod.rs

```diff
@@ -0,0 +1,240 @@
+// depth peeling, stochastic transparency, ray tracing etc.
+// This should probably be done by adding an enum to this component
+#[derive(Component, Clone, Copy, ExtractComponent)]
+pub struct OrderIndependentTransparencySettings {
+    /// Controls how many layers will be used to compute the blending.
+    /// The more layers you use the more memory it will use but it will also give better results.
+    /// 8 is generally recommended, going above 16 is probably not worth it in the vast majority of cases
+    pub layer_count: u8,
+}
+
+impl Default for OrderIndependentTransparencySettings {
+    fn default() -> Self {
+        Self { layer_count: 8 }
+    }
+}
+
+/// Plugin needed to enable Order Independent Transparency
+pub struct OrderIndependentTransparencyPlugin;
+impl Plugin for OrderIndependentTransparencyPlugin {
+    fn build(&self, app: &mut bevy_app::App) {
+        load_internal_asset!(
+            app,
+            OIT_DRAW_SHADER_HANDLE,
+            "oit_draw.wgsl",
+            Shader::from_wgsl
+        );
+
+        app.add_plugins((
+            ExtractComponentPlugin::<OrderIndependentTransparencySettings>::default(),
+            OitResolvePlugin,
+        ))
+        .add_systems(Update, check_msaa)
+        .add_systems(Last, configure_depth_texture_usages);
+
+        let Some(render_app) = app.get_sub_app_mut(RenderApp) else {
+            return;
+        };
+
+        render_app.add_systems(
+            Render,
+            prepare_oit_buffers.in_set(RenderSet::PrepareResources),
+        );
+
+        render_app
+            .add_render_graph_node::<ViewNodeRunner<OitResolveNode>>(Core3d, OitResolvePass)
+            .add_render_graph_edges(Core3d, (Node3d::MainTransparentPass, OitResolvePass));
+    }
+
+    fn finish(&self, app: &mut bevy_app::App) {
+        let Some(render_app) = app.get_sub_app_mut(RenderApp) else {
+            return;
+        };
+
+        render_app.init_resource::<OitBuffers>();
+    }
+}
+
+// WARN This should only happen for cameras with the [`OrderIndependentTransparencySettings`]
+// but when multiple cameras are present on the same window
+// bevy reuses the same depth texture so we need to set this on all cameras.
+fn configure_depth_texture_usages(mut new_cameras: Query<&mut Camera3d, Added<Camera3d>>) {
+    for mut camera in &mut new_cameras {
+        let mut usages = TextureUsages::from(camera.depth_texture_usages);
+        usages |= TextureUsages::RENDER_ATTACHMENT | TextureUsages::TEXTURE_BINDING;
+        camera.depth_texture_usages = usages.into();
+    }
+}
+
+fn check_msaa(cameras: Query<&Msaa, With<OrderIndependentTransparencySettings>>) {
+    for msaa in &cameras {
+        if msaa.samples() > 1 {
+            warn_once!(
+                "MSAA should be disabled when using Order Independent Transparency. \
+                It will cause some rendering issues on some platform. Consider using another AA method."
```

Comment:
Having msaa enabled causes a crash due to sample mismatch on depth, so we should probably panic here or forcibly disable it for the user (not sure what our policy is). The example also suggests it "doesn't work well" which is certainly true in terms of crashing but might also be misleading.

Reply:
Ah, right, I forgot about that. In my original impl I had a hack to sample the multisampled depth but it only looked correct on nvidia cards while looking weird on everything else. I removed that hack from this PR so now it just crashes. I do want to fully support MSAA eventually, but for now I'll panic here instead to keep this PR small (and because I haven't figured out a good solution yet 🙃).